pygfunction/media.py

```diff
@@ -22,63 +25,61 @@ class Fluid:
         T: float, optional
             The temperature of the fluid (in Celcius).
             Default is 20 degC.
-        P: float, optional
-            The pressure of the fluid (in Pa).
-            Default is 101325 Pa.
 
         Examples
         --------
         >>> import pygfunction as gt
         >>> T_f = 20.     # Temp at 20 C
-        >>> gage_P = 20  # PsiG
-        >>> atm_P = 14.69595
-        >>> P = (gage_P + atm_P) * 6894.75728  # Pressure in Pa
 
         >>> # complete water solution
         >>> mix = 'Water'
         >>> percent = 0
-        >>> fluid = gt.media.Fluid(mix, percent, T=T_f, P=P)
+        >>> fluid = gt.media.Fluid(mix, percent, T=T_f)
         >>> print(fluid)
 
         >>> # 20 % propylene glycol mixed with water
         >>> mix = 'MPG'
         >>> percent = 20
-        >>> fluid = gt.media.Fluid(mix, percent, T=T_f, P=P)
+        >>> fluid = gt.media.Fluid(mix, percent, T=T_f)
 
         >>> # 60% ethylene glycol mixed with water
         >>> mix = 'MEG'
         >>> percent = 60
-        >>> fluid = gt.media.Fluid(mix, percent, T=T_f, P=P)
+        >>> fluid = gt.media.Fluid(mix, percent, T=T_f)
         >>> print(fluid)
 
-        >>> # 5% methanol mixed with water water
+        >>> # 5% methanol mixed with water
         >>> mix = 'MMA'
         >>> percent = 5
-        >>> fluid = gt.media.Fluid(mix, percent, T=T_f, P=P)
+        >>> fluid = gt.media.Fluid(mix, percent, T=T_f)
         >>> print(fluid)
 
         >>> # ethanol / water
         >>> mix = 'MEA'
         >>> percent = 10
-        >>> fluid = gt.media.Fluid(mix, percent, T=T_f, P=P)
+        >>> fluid = gt.media.Fluid(mix, percent, T=T_f)
         >>> print(fluid)
     """
-    def __init__(self, mixer: str, percent: float,
-                 T: float = 20., P: float = 101325.):
-        if mixer == 'Water':
-            self.fluid_mix = mixer
-        elif mixer in ['MEG', 'MPG', 'MMA', 'MEA']:  # Expected brines
-            self.fluid_mix = f'INCOMP::{mixer}-{str(percent)}%'
+    def __init__(self, fluid_str: str, percent: float, T: float = 20.):
+        # concentration fraction
+        x_frac = percent / 100
+
+        if fluid_str.upper() == 'WATER':
+            self.fluid = Water()
+        elif fluid_str.upper() in ['PROPYLENEGLYCOL', 'MPG']:
+            self.fluid = PropyleneGlycol(x_frac)
+        elif fluid_str.upper() in ['ETHYLENEGLYCOL', 'MEG']:
+            self.fluid = EthyleneGlycol(x_frac)
+        elif fluid_str.upper() in ['METHYLALCOHOL', 'MMA']:
+            self.fluid = MethylAlcohol(x_frac)
+        elif fluid_str.upper() in ['ETHYLALCOHOL', 'MEA']:
+            self.fluid = EthylAlcohol(x_frac)
```

Comment:
I don't think it should be up to `pygfunction` to determine the fluid. Please see https://github.com/mitchute/SecondaryCoolantProps/pull/8.

Reply:
I’ve now set the command line interface of `scp` to pass through the `Fluid` class. The `Fluid` class is therefore reusable by `pygfunction` and `scp`s command line interface. The `Fluid` class provides `scp` with a polymorphic interface.

Reply:
This was addressed on the scp github.